pkg/stub/handler.go

```diff
@@ -933,11 +940,27 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 }
 
 func (h *Handler) buildSkipFilters(opcfg *v1alpha1.SamplesResource) {
+	newStreamMap := make(map[string]bool)
+	newTempMap := make(map[string]bool)
 	for _, st := range opcfg.Spec.SkippedTemplates {
 		h.skippedTemplates[st] = true
+		newTempMap[st] = true
 	}
 	for _, si := range opcfg.Spec.SkippedImagestreams {
 		h.skippedImagestreams[si] = true
+		newStreamMap[si] = true
+	}
+	//golang delete map entries while ranging is safe/supported
+	// delete entries that have been removed
+	for key, _ := range h.skippedImagestreams {
+		if _, ok := newStreamMap[key]; !ok {
+			delete(h.skippedImagestreams, key)
+		}
+	}
+	for key, _ := range h.skippedTemplates {
+		if _, ok := newTempMap[key]; !ok {
+			delete(h.skippedTemplates, key)
+		}
```

Comment:
can't you just assign h.skippedImageStreams=newStreamMap and h.skippedTemplates=newTempMap ?